### ISSUE TITLE
Missing alt text results in opaque error message

### ISSUE THREAD

**Issue by FredKSchott** [withastro/astro#8248]
### Astro info

```block
Astro version            v3.0.0-rc.7
Package manager          pnpm
Platform                 darwin
Architecture             arm64
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/tailwind, @astrojs/solid-js, @astrojs/mdx, @astrojs/sitemap, @astrojs/prefetch
```

### What browser are you using?

Chrome

### Describe the Bug

I saw this error while building astro.build repo. It didn't give me any information about how to resolve, so I ended up needing to go through the astro.build codebase manually to find any `<Image>` component without an `alt` text until I eventually found it in an MDX file.
 
<img width="1184" alt="Screen Shot 2023-08-27 at 10 06 16 AM" src="https://github.com/withastro/astro/assets/622227/dee4c878-0cfa-400f-8cdb-71b66a27f7e0">

### What's the expected result?

There should be some information in this error message to point me to the right file to fix it. 

### Link to Minimal Reproducible Example

https://github.com/withastro/astro.build/tree/astro-3-upgrade

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Princesseuh** [comment]
I think this is a larger issue with error location not getting attached properly in some specific rendering case, and not proper to this error message. Something to note is that this error message is called from "user code", and not in Astro itself, so it works a bit differently

**Comment by matthewp** [comment]
Here's a minimal example: https://stackblitz.com/edit/github-sxr1mx?file=astro.config.mjs,package.json,src%2Fpages%2Fother.mdx,src%2Fpages%2Findex.astro,src%2Fassets%2Fexample.webp&on=stackblitz

Looks like it does show up in the browser error message:

<img width="1461" alt="Screen Shot 2023-08-28 at 8 45 39 AM" src="https://github.com/withastro/astro/assets/361671/d3c1b1b1-9400-4e83-a99a-12b02413d259">

This also happens in a `.astro` file, so not related to MDX. I think this might just be a case where we never designed the CLI to give this info.

**Comment by Princesseuh** [comment]
Hmm, the CLI is definitely supposed to show the location when it has it: https://github.com/withastro/astro/blob/ed39fbf168f7a30c4fbf5b5618b70146ce5d5b3e/packages/astro/src/core/messages.ts#L219-L228

**Comment by Dunqing** [comment]
In the latest version, the error message seems good

![image](https://github.com/withastro/astro/assets/29533304/e126fc90-b6d1-4a27-975f-7dd366335101)

https://stackblitz.com/edit/github-sxr1mx-wcov2c?file=package.json

**Comment by jacobdalamb** [comment]
Can this be closed?

**Comment by FredKSchott** [comment]
Yes, I think so! Fixed in Astro 4.0
